### PR TITLE
feat: background grid

### LINKED ISSUE #202: [FEATURE]: Add a background grid effect
### Feature Description 📝

Add a component that makes it easy to add a customizable background grid.

### Screenshots 📸

_No response_

### PR DESCRIPTION
**Pull request by epoll31**
## Description

I added a simple component in the Effects group named BackgroundGrid that can be used to add a background grid to any page or any container very easily.

## Related Issue

Fixes #202 

## Proposed Changes

- added the BackgroundGrid component to `src/showcase/effects/BackgroundGrid.tsx`
- added the BackgroundGrid example to `src/showcase/ui-group/EffectCards.tsx`
  - for this, I added a check in the `data.map` that will display an Image only when `item.image` exists and will display a preview only when `item.preview` exists so that we can use components as previews instead of only images
- added the docs page to `src/app/(docs)/effects/background-grid/page.mdx`
  - I used `ComponentPreview` with the `preview` override prop so that I can customize the props like `size`, `color`, `strokeWidth`, etc. to make it fit better on the page.
- added the appropriate link to `src/components/Navigation.tsx` with the `new` tag
- added my name to the `credits.md` under this new addition

## Screenshots

Screenshot 2024-05-21 at 10 47 58 AM
Screenshot 2024-05-21 at 10 48 11 AM

## Checklist

Please check the boxes that apply:

- [x] I have rebased my branch on top of the latest `main` branch.
- [x] I have tested the changes locally
- [x] I ran `npm run build` and build is successful
- [x] I have added necessary documentation (if applicable)
- [x] I have updated the credits.md file (if applicable)

### CODE REVIEW

File: src/showcase/ui-group/EffectCards.tsx [hunk 1]
```diff
@@ -24,14 +38,17 @@ const EffectCards = () => {
...
+                {item.image && (
+                  <Image
+                    width={200}
+                    height={200}
+                    src={item.image}
+                    alt={item.title}
+                    className="h-auto w-[150px] duration-300 ease-in-out group-hover:scale-110"
+                  />
+                )}
+                {item.preview}
```
Comment (Ansub) [inline src/showcase/ui-group/EffectCards.tsx:51]: Can you add a similar logic like it's been in `ComponentCards.tsx`

**Comment by epoll31** [comment]
I've made the changes, but the new `cn` function is breaking things, so it it not ready to be merged just yet.

**Comment by epoll31** [comment]
I updated the `BackgroundGrid` component to have an elliptical fade effect now. I also removed the fade controls from the component which makes it more opinionated.

Screenshot 2024-05-30 at 11 22 58 AM

**Comment by Ansub** [comment]
can you fix the merge conflicts bro, i will merge it then

**Comment by epoll31** [comment]
Should be good now

**Pull request merged**